core/cachestub/batch_cache_stub.go

```diff
@@ -0,0 +1,57 @@
+package cachestub
+
+import (
+	"github.com/anoideaopen/foundation/proto"
+	"github.com/hyperledger/fabric-chaincode-go/shim"
+)
+
+type BatchCacheStub struct {
+	shim.ChaincodeStubInterface
+	batchCache map[string]*proto.WriteElement
+	Swaps      []*proto.Swap
+	MultiSwaps []*proto.MultiSwap
+}
+
+func NewBatchCacheStub(stub shim.ChaincodeStubInterface) *BatchCacheStub {
+	return &BatchCacheStub{
+		ChaincodeStubInterface: stub,
+		batchCache:             make(map[string]*proto.WriteElement),
+	}
+}
+
+// GetState returns state from BatchCacheStub cache or, if absent, from chaincode state
+func (bs *BatchCacheStub) GetState(key string) ([]byte, error) {
+	existsElement, ok := bs.batchCache[key]
+	if ok {
+		return existsElement.Value, nil
+	}
+	return bs.ChaincodeStubInterface.GetState(key)
```

Comment:
Could the value be put in the cache right away?

Reply:
Great point for discussion. Since this portion of the code was moved "AS IS", could we consider making changes in a separate pull request?

Reply:
I didn't see the Read cache in that file, only write cache.